File: app/src/main/java/br/com/vicentec12/mygames/data/source/DataModule.java
package br.com.vicentec12.mygames.data.source;

import android.content.Context;

import androidx.room.Room;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

import static br.com.vicentec12.mygames.data.source.AppDatabase.MIGRATION_1_2;

@Module
public abstract class DataModule {

    @Singleton
    @Provides
    static AppDatabase provideLocalDatabase(Context mContext) {
        return Room.databaseBuilder(mContext, AppDatabase.class, AppDatabase.DATABASE_NAME)
                .addMigrations(MIGRATION_1_2).fallbackToDestructiveMigration().build();
    }

}

File: app/src/main/java/br/com/vicentec12/mygames/data/source/DataModule.kt
package br.com.vicentec12.mygames.data.source

import android.content.Context
import androidx.room.Room
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object DataModule {

    @Provides
    @Singleton
    fun providesAppDatabase(mContext: Context): AppDatabase {
        return Room.databaseBuilder(mContext, AppDatabase::class.java, AppDatabase.DATABASE_NAME)
                .build()
    }

}